sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala

```diff
@@ -226,4 +227,22 @@ object StaticSQLConf {
       .version("3.0.0")
       .intConf
       .createWithDefault(100)
+
+  val SPARK_SCRATCH_DIR =
+    buildStaticConf("spark.scratchdir")
+      .doc("Scratch space for Spark temporary table and so on. Similar with hive.exec.scratchdir")
```

Comment:
let's not bring up hive here. Slowly nobody will care about Hive.

Also this should be spark.sql.scratchdir?